Grillbot/Modules/DuckModule.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Discord.Commands;
+using Grillbot.Extensions;
+using Grillbot.Extensions.Discord;
+using Grillbot.Models.Duck;
+using Grillbot.Models.Embed;
+using Grillbot.Services;
+using Grillbot.Services.Config.Models;
+using Grillbot.Services.Preconditions;
+using Microsoft.Extensions.Options;
+
+namespace Grillbot.Modules
+{
+    [Name("Stav Kachny")]
+    [Group("kachna")]
+    [RequirePermissions]
+    public class DuckModule : BotModuleBase
+    {
+        private BotLoggingService Logger { get; set; }
+
+        public DuckModule(IOptions<Configuration> config, BotLoggingService logger) : base(config)
+        {
+            Logger = logger;
+        }
+
+        [Command("", true)]
+        public async Task GetDuckInfoAsync()
+        {
+            var config = GetMethodConfig<DuckConfig>("kachna", "");
+
+            await DoAsync(async () =>
+            {
+                var client = new HttpClient();
+                client.BaseAddress = new Uri(config.IsKachnaOpenApiBase);
+                HttpResponseMessage resp;
+                
+                try
+                {
+                    resp = await client.GetAsync("api/duck/currentState");
+                }
+                catch
+                {
+                    await Logger.WriteAsync(
+                        "Request na IsKachnaOpen skončil špatně (nepodařilo se navázat spojení nebo jiná výjimka.)",
+                        "DUCK");
+
+                    throw new ArgumentException("Nepodařilo se zjistit stav Kachny. Zkus " +
+                                                config.IsKachnaOpenApiBase);
+                }
+
+                if (!resp.IsSuccessStatusCode)
+                {
+                    await Logger.WriteAsync(
+                        $"Request na IsKachnaOpen skončil špatně (HTTP {(int) resp.StatusCode}).\n{await resp.Content.ReadAsStringAsync()}",
+                        "DUCK");
+
+                    throw new ArgumentException("Nepodařilo se zjistit stav Kachny. Zkus " +
+                                                config.IsKachnaOpenApiBase);
+                }
+
+                var dto = await resp.Content.ReadAsAsync<CurrentState>();
+                var embed = this.MakeEmbed(dto, config.EnableBeersOnTap);
+                await this.ReplyAsync(embed: embed.Build()).ConfigureAwait(false);
+            }).ConfigureAwait(false);
+        }
+
+
+        private BotEmbed MakeEmbed(CurrentState state, bool enableBeers)
+        {
+            var user = Context.Guild == null ? Context.User : Context.Guild.GetUser(Context.User.Id);
```

Comment:
Use the extension method  Context.Guild.GetUserFromGuildAsync to avoid cache problems.

Reply:
Done. Also added an overload of the extension method that accepts an ulong, so that I don't have to convert the ulong value to string and back again.